JavaScript/chapter01/p08_zero_matrix/avc278.js

```diff
@@ -0,0 +1,94 @@
+// Zero Matrix: Write an algorithm such that if an element in an MxN matrix is 0, its entire row and columns are set to 0.
+
+const assert = require("assert");
+
+/**
+ * Overwrite rows and columns who have at least 1 zero in them with zeros.
+ * @param  {Array.<number[]>} matrix input 2d array to mutate
+ * @return {null}
+ *
+ * Okay, so I have a jank solution that assumes no element in the array is currently -999.
```

Comment:
haha, i laughed out loud when i read this.

ahem. would it be possible to replace the "magic number" -999 by something that cannot be confused with a valid number? say... [`NaN`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/NaN)? it is still represented as a [`Number`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number) (so it does not have the undesirable property of `undefined` or `null` that the type must be relaxed to be able to express this constant).

normally whenever there's a need to have a sentinel value, it should be chosen in a way that it is not representable as a valid item in the range of the type. for floating-point numbers, `NaN` is a good candidate, but typically `null` or `undefined` should be preferred.

Reply:
That makes a lot of sense. At work, my boss and I agreed to hard code years to only be four places because "once it's between the year 9999 and 10000, it's no longer something we have to worry about 😉 "